### ISSUE TITLE
3 state buttons - ignore, include, exclude

### ISSUE THREAD

**Issue by aisbergde** [ransome1/sleek#414]
I want to request an easy way to exclude some context or projects from the item list. Then I could for example easy exclude "@Bookmark", "@Someone", "begin in future" and so on.

For easy doing this I suggest to use 3 state buttons instead of 2 state buttons:
- ignore
- include
- exclude (this will be an additional state)

This way the condition for the excluded items are optical shown and can be easy changed.

Additionaly I suggest these kind of 3 state buttons for some or all settings, which are currently separate filter settings using switches on another page. Instead of using buttons only for projects and context, they could be used more generally. This allow easy seeing and changing them in a consistent way, in one place, together with other buttons:

- done tasks
- begin date in future
- due date in the past
-... 

Currently I miss a lot of the flexibility in filtering, which provides SimpleTask on Android. These 3 state buttons could add a lot of this flexibility. And if sets of button states could be easy saved and restored, then this would be an equivalent to the stored lists in Simpletask.

If it is too early to use this as feature request, then just convert it into a discussion.

**Comment by ransome1** [comment]
A first draft of how this kind of feature could look like in the future has been added to this early developer preview: https://github.com/ransome1/sleek/releases/tag/v2.0.0-dev1

- A regular click on a filter will reduce the todos to those, which contain the filter.
- If you press and hold Ctrl or CMD and hover over a filter, a closed eye should appear. If you click on this filter now, it will exclude the todos, which contain the filter.

As said, it is a very early developer preview, mostly aimed for rewriting the whole app. I added this increment as it was a low hanging fruit. One that needs evaluation and testing.

![Bildschirmfoto 2023-07-19 um 16 52 10](https://github.com/ransome1/sleek/assets/11188741/81565a7c-1c40-405e-8891-77c0b3e5d481)
